DCOS-45978: Add label tooltip to text input component

GeorgiSTodorov opened the pull request:
Add an option for TextInput to have a tooltip with an icon.

Closes https://jira.mesosphere.com/browse/DCOS-45978

## Testing
Verify that a `TextInput` with `tooltipContent` works and so do the other `TextInput`s.
Please review critically.

## Screenshots
![Screenshot from 2019-04-23 10-48-30](https://user-images.githubusercontent.com/40791275/56563964-b748d380-65b5-11e9-80e8-f5cb21693663.png)

![Screenshot from 2019-04-25 09-36-44](https://user-images.githubusercontent.com/40791275/56729070-f9f8e000-675c-11e9-954f-6121978584ab.png)

![Screenshot from 2019-04-23 10-49-33](https://user-images.githubusercontent.com/40791275/56564067-f119da00-65b5-11e9-94f6-4dc08241dc96.png)

GeorgiSTodorov commented on packages/textInput/components/TextInput.tsx:
```diff
@@ -90,7 +103,7 @@ export class TextInput<
           margin("bottom", "xxs"),
           textWeight("medium"),
-          display("block"),
+          this.props.tooltipText ? display("inline-block") : display("block"),
```
This was needed for the tooltip to be on the same line.

mperrotti commented on packages/textInput/components/TextInput.tsx:
```diff
@@ -65,13 +76,15 @@ export class TextInput<
     return (
       <div {...containerProps}>
         {labelContent}
+        {tooltipContent}
```
We get nicer vertical alignment if we use a flexbox container to wrap both the `<label>` and the `<Tooltip>`, and move the bottom margin from the `<label>` to the flexbox container:
```
<div className={cx(margin("bottom", "xxs"), flex({ align: "center" }))}>
    {labelContent}
    {tooltipContent}
</div>
```

Then, we can also get rid of the ternary in `getLabelContent()`: https://github.com/dcos-labs/ui-kit/pull/327/files#diff-d6ddbf113c06302e95d5b248c029f35fR106

mperrotti commented on packages/textInput/components/TextInput.tsx:
```diff
@@ -177,6 +192,29 @@ export class TextInput<
+              size={iconSizeXs}
+            />
+          }
+          id="tooltipTextinput"
```
We'll get duplicate IDs if we just use a string for this. We could use the `id` prop to make the tooltip IDs unique

mperrotti commented:
One other thing to consider:
I've seen some long pieces of text in some of the field label tooltips in dcos-ui. Should we pass a default `maxWidth` to the `Tooltip` component here? 200px feels reasonable

mperrotti commented on packages/textInput/components/TextInput.tsx:
```diff
@@ -177,6 +192,29 @@ export class TextInput<
+    }
+
+    return (
+      <span className={cx(margin("left", "xs"))}>
```
No need for the `cx` here. You can just do `className={margin("left", "xs")}`

mperrotti commented on packages/textInput/components/TextInput.tsx:
```diff
@@ -177,6 +192,29 @@ export class TextInput<
+              size={iconSizeXs}
+            />
+          }
+          id={this.props.id}
```
```suggestion
          id={`labelTooltip-${this.props.id}`}
```

mperrotti replied:
Otherwise the tooltip will have the same ID as the text input

mperrotti commented on packages/textInput/components/TextInput.tsx:
```diff
@@ -177,6 +192,29 @@ export class TextInput<
+            />
+          }
+          id={this.props.id}
+        >
```
```suggestion
          maxWidth={200}
        >
```

mperrotti replied:
This would apply the default `maxWidth` I mentioned :)

mperrotti commented:
Also we don't need these classes anymore. Once we remove them, the vertical alignment gets better:
https://github.com/dcos-labs/ui-kit/pull/327/files#diff-d6ddbf113c06302e95d5b248c029f35fR106
https://github.com/dcos-labs/ui-kit/pull/327/files#diff-d6ddbf113c06302e95d5b248c029f35fR106

GeorgiSTodorov commented:
@mperrotti 
> Also we don't need these classes anymore. Once we remove them, the vertical alignment gets better:
https://github.com/dcos-labs/ui-kit/pull/327/files#diff-d6ddbf113c06302e95d5b248c029f35fR106
https://github.com/dcos-labs/ui-kit/pull/327/files#diff-d6ddbf113c06302e95d5b248c029f35fR106

You posted the same line of code twice. Did you mean to do that?

GeorgiSTodorov commented:
@mperrotti , I addressed your feedback. Regarding the comment about removing classes, you posted the same line twice, so I only removed one class. Are there any other classes that need to be removed?

mperrotti commented:
Oops, I did not mean that. I meant `display("block")`

mperrotti commented:
Snapshots just need to be updated :)

GeorgiSTodorov commented:
Merging the latest master was causing some snapshots, unrelated to this PR, to fail. I fixed them now. @pierrebeitz @TattdCodeMonkey , please re-review.]
![Screenshot from 2019-05-07 11-25-59](https://user-images.githubusercontent.com/40791275/57285225-6956c400-70bb-11e9-9a0f-fed676380518.png)
